community/kernel/src/test/java/org/neo4j/kernel/impl/util/dbstructure/CineastsDbStructure.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.util.dbstructure;
+
+import org.neo4j.helpers.collection.Visitable;
+import org.neo4j.kernel.impl.util.dbstructure.DbStructureVisitor;
+
+import org.neo4j.kernel.api.constraints.UniquenessConstraint;
+import org.neo4j.kernel.api.index.IndexDescriptor;
+
+//
+// GENERATED FILE. DO NOT EDIT. 
```

Comment:
The whole point of this PR is to check in the generated classes instead of checking in the actual datasets (which would be prohibitively huge) so that we may depend on realistic scenarios in Cypher tests and also make it easy to investigate adding additional statistics easily.  The only alternative I can see would have been to come up with our own serialization format for this. That would require more work and I don't see the benefit of it.